daemon/daemon_cleanup.go

```diff
@@ -0,0 +1,178 @@
+// Copyright 2017 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"strings"
+
+	"github.com/cilium/cilium/daemon/defaults"
+	"github.com/cilium/cilium/pkg/bpf"
+	"github.com/vishvananda/netlink"
+)
+
+const (
+	ciliumPrefix = "cilium"
+	tunnelMap    = "tunnel_endpoint_map"
+	cniConfig    = "/etc/cni/net.d/10-cilium-cni.conf"
+)
+
+func runCleanup() {
+	// Abort if the agent is running
+	if _, err := os.Stat(defaults.SockPath); !os.IsNotExist(err) {
+		fmt.Fprintf(os.Stderr, "Agent should not be running when cleaning up\n"+
```

Comment:
Is the socket always removed when the agent crashes for example? What about creating a `.pid` file?

Reply:
Using pidfile, thanks.